sqlmesh/core/engine_adapter/base.py

```diff
@@ -303,34 +347,22 @@ def _create_table_from_columns(
         )
         self._create_table(schema, None, exists=exists, **kwargs)
 
-    def _create_table_from_query(
+    def _create_table_from_source_queries(
         self,
         table_name: TableName,
-        query: Query,
-        exists: bool = True,
-        replace: bool = False,
-        **kwargs: t.Any,
-    ) -> None:
-        table = exp.to_table(table_name)
-        self._create_table(table, query, exists=exists, replace=replace, **kwargs)
-
-    def _create_table_from_df(
-        self,
-        table_name: TableName,
-        df: DF,
+        source_queries: t.List[SourceQuery],
         columns_to_types: t.Optional[t.Dict[str, exp.DataType]] = None,
         exists: bool = True,
         replace: bool = False,
         **kwargs: t.Any,
     ) -> None:
-        if not isinstance(df, pd.DataFrame):
-            raise ValueError("df must be a pandas DataFrame")
-        columns_to_types = columns_to_types or columns_to_types_from_df(df)
-        with self.transaction():
-            if replace:
-                self.drop_table(table_name)
-            self._create_table_from_columns(table_name, columns_to_types, **kwargs)
-            self._insert_append_pandas_df(table_name, df, columns_to_types)
+        table = exp.to_table(table_name)
+        multiple_queries = len(source_queries) > 1
```

Comment:
why not just

with self.transaction():
  for i, query in enumerate(source_queries):
        if i == 0:
          self._create_table()
      else: 
          self._insert_append(query)

Reply:
This is functionally the same and currently we don't have an insert/append that takes in a `Query` object.